### PR TITLE
#160702852 Implement signup endpoint

### PR DESCRIPTION
**Pull request by sudhons**
#### What does this PR do?
create the endpoint for new user to signup

#### Description of Task to be completed?
- Install bcryptjs
- write tests for the endpoint
- write validation for the endpoint
- write SQL queries to save a user to database
- write the controller to create the user
- hash user's password

#### What are the relevant pivotal tracker stories?
[#160702852](https://www.pivotaltracker.com/story/show/160702852)
 [#160720394](https://www.pivotaltracker.com/story/show/160720394)

### CODE REVIEW

File: API/src/queries/createTables.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    first_name VARCHAR(40) NOT NULL,
+    last_name VARCHAR(40) NOT NULL,
+    email VARCHAR(100) UNIQUE NOT NULL,
+    password VARCHAR(100) NOT NULL
+  );
+
+  INSERT INTO users(first_name, last_name, email, password, user_role)
+    VALUES(
+      'Oluwaseun', 'Sunday', 'sudhons@yahoo.com',
+      '${bcrypt.hashSync('sudhons', 10)}', 'admin'
```
Comment (dogoyaro) [inline API/src/queries/createTables.js:30]: Sure you are calling this function the right way. The documentation specifies a way to generate the salt used in the hash. Is there a reason why you have chosen not to use that?

File: API/test/app.test.js [hunk 2]
```diff
@@ -46,10 +54,81 @@ describe('App', () => {
...
+        .post('/api/v1/auth/signup')
+        .send({
+          firstName,
+          lastName,
+          email,
+          password
+        })
+        .end((error, response) => {
+          assert.strictEqual(response.status, 201);
+          assert.hasAllKeys(response.body, ['status', 'message', 'data']);
```
Comment (dogoyaro) [inline API/test/app.test.js:128]: Perhaps you might want to assert that the values are what you expected alongside what you have now.

File: API/src/queries/createTables.js (hunk 1 above)
Comment (sudhons) [reply API/src/queries/createTables.js:30]: The format is also available in the documentation. It automatically generate the salt, here is the [link](https://www.npmjs.com/package/bcryptjs#usage---sync)

File: API/test/app.test.js (hunk 2 above)
Comment (sudhons) [reply API/test/app.test.js:128]: Ok

File: API/src/queries/createTables.js (hunk 1 above)
Comment (dogoyaro) [reply API/src/queries/createTables.js:30]: Awesome!

**Pull request merged**